src/python/grpcio_tests/tests_aio/unit/client_unary_stream_interceptor_test.py

```diff
@@ -0,0 +1,404 @@
+
+_SHORT_TIMEOUT_S = datetime.timedelta(seconds=1).total_seconds()
+
+_LOCAL_CANCEL_DETAILS_EXPECTATION = 'Locally cancelled by application!'
+_NUM_STREAM_RESPONSES = 5
+_REQUEST_PAYLOAD_SIZE = 7
+_RESPONSE_PAYLOAD_SIZE = 7
+_RESPONSE_INTERVAL_US = int(_SHORT_TIMEOUT_S * 1000 * 1000)
+
+
+class _ResponseIterator:
+
+    def __init__(self, response_iterator):
+        self._response_cnt = 0
+        self._response_iterator = response_iterator
+
+    async def _forward_responses(self):
+        async for response in self._response_iterator:
+            self._response_cnt += 1
+            yield response
+
+    def __aiter__(self):
+        return self._forward_responses()
+
+    @property
+    def response_cnt(self):
+        return self._response_cnt
+
+
+def _inject_callbacks(call):
+    first_callback_ran = asyncio.Event()
+
+    def first_callback(call):
+        # Validate that all resopnses have been received
+        # and the call is an end state.
+        assert call.done()
+        first_callback_ran.set()
+
+    second_callback_ran = asyncio.Event()
+
+    def second_callback(call):
+        # Validate that all resopnses have been received
+        # and the call is an end state.
+        assert call.done()
+        second_callback_ran.set()
+
+    call.add_done_callback(first_callback)
+    call.add_done_callback(second_callback)
+
+    async def validation():
+        await asyncio.wait_for(
+            asyncio.gather(first_callback_ran.wait(),
+                           second_callback_ran.wait()),
+            test_constants.SHORT_TIMEOUT)
+
+    return validation()
+
+
+class _UnaryStreamInterceptorEmpty(aio.UnaryStreamClientInterceptor):
+
+    async def intercept_unary_stream(self, continuation, client_call_details,
+                                     request):
+        return await continuation(client_call_details, request)
+
+
+class _UnaryStreamInterceptorWith_ResponseIterator(
+        aio.UnaryStreamClientInterceptor):
+
+    def __init__(self):
+        self.response_iterator = None
+
+    async def intercept_unary_stream(self, continuation, client_call_details,
+                                     request):
+        call = await continuation(client_call_details, request)
+        self.response_iterator = _ResponseIterator(call)
+        return self.response_iterator
+
+
+class TestUnaryStreamClientInterceptor(AioTestBase):
+
+    async def setUp(self):
+        self._server_target, self._server = await start_test_server()
+
+    async def tearDown(self):
+        await self._server.stop(None)
+
+    async def test_intercepts(self):
+        for interceptor_class in (_UnaryStreamInterceptorEmpty,
+                                  _UnaryStreamInterceptorWith_ResponseIterator):
+
+            with self.subTest(name=interceptor_class):
+                interceptor = interceptor_class()
+
+                request = messages_pb2.StreamingOutputCallRequest()
+                for _ in range(_NUM_STREAM_RESPONSES):
```

Comment:
Nit: You can use `extend` instead of `append`:
```python
reqest.response_parameters.extend(
  [messages_pb2.ResponseParameters(size=_RESPONSE_PAYLOAD_SIZE)] * _NUM_STREAM_RESPONSES)
```

Reply:
Yeps, I will change it to extend.